fix compilation error: error: 'H5D_MPIO_FILTERS' undeclared
internal constant H5D_MPIO_FILTERS was renamed to H5D_MPIO_NO_COLLECTIVE_MAX_CAUSE in HDF5 1.10.2
see also https://github.com/hhoeflin/hdf5r/issues/97
author: Kenneth Hoste (HPC-UGent)
diff -ru hdf5r.orig/MD5 hdf5r/MD5
--- hdf5r.orig/MD5	2017-10-25 11:45:44.000000000 +0200
+++ hdf5r/MD5	2018-09-18 19:06:10.131472405 +0200
@@ -179,7 +179,7 @@
 ca5d483c11c1eabe73aa58e464e003ed *src/1_10_0/Wrapper_auto_H5Z.h
 786e8111cf97a17f3e186c23138a3a51 *src/1_10_0/const_export.c
 197c5a485540fb6a9b10b881b00805cc *src/1_10_0/const_export.h
-a0febd83da74edb5693bccd325ab9d79 *src/1_10_0/datatype_export.c
+54cfea23947efa6f83952e934ecdd811 *src/1_10_0/datatype_export.c
 93630a5e8c69fb2b041b0a60d04ef8cf *src/1_10_0/datatype_export.h
 0ba59b04936519e01219d12465a35b85 *src/1_10_0/export_auto.h
 4cf5cc29d1987c24b8a34a7f08f7574c *src/1_8_12/HelperStructs.h
diff -ru hdf5r.orig/src/1_10_0/datatype_export.c hdf5r/src/1_10_0/datatype_export.c
--- hdf5r.orig/src/1_10_0/datatype_export.c	2017-10-23 21:39:29.000000000 +0200
+++ hdf5r/src/1_10_0/datatype_export.c	2018-09-18 19:05:00.690794603 +0200
@@ -1474,7 +1474,7 @@
   return(dtype_id);
 }
 
-/* typedef enum H5D_mpio_no_collective_cause_t { H5D_MPIO_COLLECTIVE = 0x00, H5D_MPIO_SET_INDEPENDENT = 0x01, H5D_MPIO_DATATYPE_CONVERSION = 0x02, H5D_MPIO_DATA_TRANSFORMS = 0x04, H5D_MPIO_MPI_OPT_TYPES_ENV_VAR_DISABLED = 0x08, H5D_MPIO_NOT_SIMPLE_OR_SCALAR_DATASPACES = 0x10, H5D_MPIO_NOT_CONTIGUOUS_OR_CHUNKED_DATASET = 0x20, H5D_MPIO_FILTERS = 0x40} H5D_mpio_no_collective_cause_t; */
+/* typedef enum H5D_mpio_no_collective_cause_t { H5D_MPIO_COLLECTIVE = 0x00, H5D_MPIO_SET_INDEPENDENT = 0x01, H5D_MPIO_DATATYPE_CONVERSION = 0x02, H5D_MPIO_DATA_TRANSFORMS = 0x04, H5D_MPIO_MPI_OPT_TYPES_ENV_VAR_DISABLED = 0x08, H5D_MPIO_NOT_SIMPLE_OR_SCALAR_DATASPACES = 0x10, H5D_MPIO_NOT_CONTIGUOUS_OR_CHUNKED_DATASET = 0x20, H5D_MPIO_NO_COLLECTIVE_MAX_CAUSE = 0x40} H5D_mpio_no_collective_cause_t; */
 hid_t create_DT_H5D_mpio_no_collective_cause_t(void) {
   enum H5D_mpio_no_collective_cause_t myenum;
   hid_t base_type = get_h5_equiv(sizeof(enum H5D_mpio_no_collective_cause_t), issigned(enum H5D_mpio_no_collective_cause_t));
@@ -1493,8 +1493,8 @@
   H5Tenum_insert(dtype_id, "H5D_MPIO_NOT_SIMPLE_OR_SCALAR_DATASPACES", &myenum);
   myenum = H5D_MPIO_NOT_CONTIGUOUS_OR_CHUNKED_DATASET;
   H5Tenum_insert(dtype_id, "H5D_MPIO_NOT_CONTIGUOUS_OR_CHUNKED_DATASET", &myenum);
-  myenum = H5D_MPIO_FILTERS;
-  H5Tenum_insert(dtype_id, "H5D_MPIO_FILTERS", &myenum);
+  myenum = H5D_MPIO_NO_COLLECTIVE_MAX_CAUSE;
+  H5Tenum_insert(dtype_id, "H5D_MPIO_NO_COLLECTIVE_MAX_CAUSE", &myenum);
   return(dtype_id);
 }
 
